routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
contexts have a `Value` field, which gives you a key value mapping that is tied to a particular chain of contexts. With the goprocess thing in the middle, those value requests dont get propagated up the chain like they would if we exclusively used contexts.